src/lib/client.js

```diff
@@ -3,23 +3,27 @@
 const assert = require('assert')
 const WebSocket = require('ws')
 const debug = require('debug')('ilp-plugin-payment-channel-framework:client')
-const url = require('url')
+const { URL } = require('url')
 
 module.exports = class BtpClient {
-  constructor ({ server, secret, plugin, insecure }) {
-    this._server = server
+  constructor ({ btpUri, plugin }) {
     this._plugin = plugin
 
-    // The server URI must follow the format: btp+wss://host:port/path
-    // See also: https://github.com/interledger/interledger/wiki/Interledger-over-BTP
-    const parsedServer = url.parse(server)
-    assert(parsedServer.protocol.startsWith('btp+'), 'server uri must start with "btp+"')
-    this._wsUri = (insecure ? 'ws://' : 'wss://') + parsedServer.host + parsedServer.path
-    this._secret = (parsedServer.auth && parsedServer.auth.split(':')[1]) || secret
-
-    if (!secret) {
-      throw new Error('secret must be provided to BTP Client in the URL or in the configuration')
-    }
+    // The BTP URI must follow one of the following formats:
+    // btp+wss://auth_username:auth_token@host:port/path
+    // btp+wss://auth_username:auth_token@host/path
+    // btp+ws://auth_username:auth_token@host:port/path
+    // btp+ws://auth_username:auth_token@host/path
+    // See also: https://github.com/interledger/rfcs/pull/300
+    const parsedBtpUri = new URL(btpUri)
+    this._authUsername = parsedBtpUri.username
+    this._authToken = parsedBtpUri.password
+    parsedBtpUri.username = ''
+    parsedBtpUri.password = ''
+    // Note that setting the parsedBtpUri.protocol does not work as expected,
+    // so removing the 'btp+' prefix from the full URL here:
+    assert(parsedBtpUri.toString().startsWith('btp+'), 'server uri must start with "btp+"')
+    this._wsUri = parsedBtpUri.toString().substring('btp+'.length)
```

Comment:
if we just take a substring of the btp URI, it will still contain the auth details in the URI.

Reply:
no, because of https://github.com/interledgerjs/ilp-plugin-payment-channel-framework/pull/27/files/ab763b5aae14e0effaaff8d2fe9e0cc7550812a3#diff-976824b6113e95afacd333cbfab6d019R22